applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,503 @@
+    using LocalSystemVectorType    = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType    = typename BaseType::LocalSystemMatrixType;
+    using TSystemMatrixPointerType = typename BaseType::TSystemMatrixPointerType;
+    using TSystemVectorPointerType = typename BaseType::TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * Default constructor
+     * @param rModelPart The model part of the problem
+     * @param pScheme The integration scheme
+     * @param pNewLinearSolver The linear solver employed
+     * @param pNewConvergenceCriteria The convergence criteria employed
+     * @param pNewBuilderAndSolver The builder and solver
+     * @param rParameters The parameters for the strategy
+     * @param Beta The Newmark Beta parameter
+     * @param Gamma The Newmark Gamma parameter
+     * @param MaxIterations The maximum number of non-linear iterations to be considered when solving the problem
+     * @param CalculateReactions The flag for the reaction calculation
+     * @param MoveMeshFlag The flag that allows to move the mesh
+     */
+    explicit GeoMechanicNewtonRaphsonStrategyLinearElasticDynamic(
+        ModelPart&                                 rModelPart,
+        typename TSchemeType::Pointer              pScheme,
+        typename TLinearSolver::Pointer            pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer    pNewBuilderAndSolver,
+        Parameters&                                rParameters,
+        int                                        MaxIterations      = 30,
+        bool                                       CalculateReactions = false,
+        bool                                       MoveMeshFlag       = false)
+        : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(
+              rModelPart, pScheme, pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, false, MoveMeshFlag)
+    {
+        // new constructor
+    }
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY;
+
+        BaseType::Initialize();
+
+        // Note that FindNeighbourElementsOfConditionsProcess and DeactivateConditionsOnInactiveElements are required to be perfomed before initializing the System and State
+        // this means that these operations are done twice in the GeomechanicsSolver in python
+        auto find_neighbour_elements_of_conditions_process =
+            FindNeighbourElementsOfConditionsProcess(BaseType::GetModelPart());
+        find_neighbour_elements_of_conditions_process.Execute();
+
+        auto deactivate_conditions_on_inactive_elements_process =
+            DeactivateConditionsOnInactiveElements(BaseType::GetModelPart());
+        deactivate_conditions_on_inactive_elements_process.Execute();
```

Comment:
And a similar suggestion here:
```suggestion
        DeactivateConditionsOnInactiveElements{BaseType::GetModelPart()}.Execute();
```